### PR TITLE
[4.0] Updating PHPUnit to 8.0

### PR DESCRIPTION
**Pull request by Hackwar**
This updates PHPUnit to version 8.0. There are still a few warnings about deprecated code. If someone wants to work on that, be my guest.

This branch also contains the changes from #24970, since Codeception has to be updated to 3.0 as well to work together.

### CODE REVIEW

File: tests/Unit/Libraries/Cms/VersionTest.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ public function testIsCompatible()
 	 */
 	public function testGetHelpVersion()
 	{
-		$this->assertInternalType('string', $this->version->getHelpVersion());
+		$this->assertIsString('string', $this->version->getHelpVersion());
```
Comment (wilsonge) [inline tests/Unit/Libraries/Cms/VersionTest.php]: seems wrong

File: composer.json [hunk 2]
```diff
@@ -99,14 +99,14 @@
...
-        "phpunit/phpunit": "~6.0",
+        "phpunit/phpunit": "~8.0",
         "joomla/mediawiki": "dev-master",
         "friendsofphp/php-cs-fixer": "~2.12",
         "squizlabs/php_codesniffer": "~1.5",
         "joomla-projects/joomla-browser": "~4.0@dev",
         "joomla-projects/robo-joomla": "dev-develop",
         "joomla-projects/selenium-server-standalone": "~v3",
-        "codeception/codeception": "~2.3",
+        "codeception/codeception": "~3.0",
```
Comment (wilsonge) [inline composer.json]: Is this upgrade really required for phpunit 8 upgrade?

Comment (Hackwar) [reply composer.json]: Codeception in 3.0 is compatible with PHPUnit 8. The update to 3.0 is a separate PR though.

File: tests/Unit/Libraries/Cms/VersionTest.php (hunk 1 above)
Comment (Hackwar) [reply tests/Unit/Libraries/Cms/VersionTest.php]: Fixed.

**Comment by rdeutz** [comment]
looks good to me

**Comment by HLeithner** [comment]
thx

**Pull request merged**